components/basecomponents/MemoryGame/MemoryGameCore.tsx

```diff
@@ -0,0 +1,304 @@
+    };
+  }, []);
+
+  const newGame = () => {
+    console.log('new game');
+    // prepare and shuffle cards, pick 8 cards
+    const pickedCards = cardsData.sort(() => Math.random() - 0.5).slice(0, 8);
+    const coloredCards = addBackroundColor(pickedCards) as (CardData & { color: string })[];
+
+    setCards(
+      [...coloredCards, ...coloredCards]
+        .sort(() => Math.random() - 0.5)
+        .map((card) => ({ ...card, image: `/kids/${card.image}.svg`, id: Math.random(), flipped: false }))
+    );
+    setSelectedCards({ first: null, second: null });
+    // clearTimers();
+  };
+
+  const flipCard = (cardToFlip: Card) => {
+    setCards((cards) => cards.map((card) => (card.id === cardToFlip.id ? { ...card, flipped: !card.flipped } : card)));
+    cardFlipSound.play();
+  };
+
+  const selectCard = (card: Card) => {
+    if (controlsDisabled) return;
+
+    const { first, second } = selectedCards;
+    if (first === null && !card.flipped) {
+      setSelectedCards({ ...selectedCards, first: card });
+    } else if (second === null && !card.flipped) {
+      setSelectedCards({ ...selectedCards, second: card });
+    }
+  };
+
+  // handle selected cards
+  useEffect(() => {
+    const { first, second } = selectedCards;
+
+    const bothCardsSelected = first !== null && second !== null;
+
+    if (bothCardsSelected) {
+      setScene(Scene.secondCardSelected);
+      return;
+    }
+    if (first !== null) {
+      setScene(Scene.firstCardSelected);
+    }
+  }, [selectedCards]);
+
+  // resolve game states
+  useEffect(() => {
+    const sceneActions: Record<Scene, () => void> = {
+      init: () => {
+        // begin new game automaticaly
+        setScene(Scene.begin);
+      },
+      begin: () => {
+        // new game
+        newGame();
+        // disable controls
+        setControlsDisabled(true);
+        // play css animations
+        setTimer(() => {
+          setScene(Scene.game);
+        }, 1000);
+      },
+      game: () => {
+        // enable controls
+        setControlsDisabled(false);
+      },
+      firstCardSelected: () => {
+        // disable controls
+        setControlsDisabled(true);
+        // play css animations and sounds
+        const { first: card } = selectedCards;
+        flipCard(card!); // 0.3s
+        playCardPhraseOtherLang(card!);
+
+        setTimer(() => {
+          setScene(Scene.game);
+        }, 600); // reduced for fastrer UX
+      },
+      secondCardSelected: async () => {
+        // disable controls
+        setControlsDisabled(true);
+        // play css animations and sounds
+        const { second: card } = selectedCards;
+        flipCard(card!); // 0.3s
+
+        await playCardPhraseCurrentLang(card!);
+        setScene(Scene.resolveCards);
+      },
+      resolveCards: () => {
+        const { first, second } = selectedCards;
+        const cardsMatch = first?.image === second?.image;
+
+        if (cardsMatch) {
+          console.log('cards match');
```

Comment:
If you want to keep the console logs here for debugging purposes, disable the relevant eslint rule just for this file by putting `/* eslint-disable no-console */` on the top.